diagnostic.md

```diff
@@ -6,70 +6,70 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What are two ways of telling rails where to route a GET request to '/books/4'?
 
 ```rb
-// your response here
+index or show using the URL/books/4
 ```
 
 In what file does the above code go?
 
 ```md
-// your response here
+controller script
 ```
 
 In your own words, what does the command `bundle exec rake db:migrate` do?
 
 ```md
-// your response here
+it migrates the rails api to
 ```
 
 What is the difference between `rails g scaffold things` and
 `rails g controller things`?
 
 ```md
-// your response here
+Scaffold will generate the entire rails database structure.  Controller only the controller files.
 ```
 
 What is the job of the `serializer`?
 
 ```md
-// your response here
+Serializers replace all the views.
```

Comment:
The serializer allows you to filter certain data out of your response. The data still exists, it's just not shown in the response to users.